cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -161,7 +169,7 @@ type_id to_type_id(SchemaElement const& schema,
   switch (physical) {
     case BOOLEAN: return type_id::BOOL8;
     case INT32: return type_id::INT32;
-    case INT64: return type_id::INT64;
+    case INT64: return arrow.value_or(type_id::INT64);
```

Comment:
This will be non-obvious outside of the context of this PR. Along with the name change suggested above, perhaps this should be done before hitting the switch. Something like:
```c++
  if (schema.arrow_duration_type.has_value() and physical == INT64) {
    return *schema.arrow_duration_type;
  }
```
Extra credit: can we change `physical` to `physical_type`?

Reply:
All done!